### PR TITLE
Calendar filter fix: no feedback for empty dates

### LINKED ISSUE #344: Calendar filter fix: no feedback for empty dates
**Issue:** Calendar filter fix - no feedback for empty dates

When a user selects a date that has no events, the page simply shows an empty state without any user feedback. This can be confusing as users may not know if the filter is working correctly or if there's a loading issue. We need to add a snackbar to inform the user that there's no event on the date they selected.

### PR DESCRIPTION
**Pull request by rita-naimi**
> [!important] 
> Line coverage is above 90%.

## What

Adds a snackbar notification when a user selects a date from the calendar that has no associated events. The snackbar displays "No events on this date" and auto-dismisses after 2.5 seconds.

## Why

Previously, when selecting a date with no events, the page would show an empty state without any feedback to the user. The snackbar provides immediate feedback that the selected date has no events, improving the user experience.

## How

- Added `SnackbarHost` to the HomeScreen Scaffold
- Added state tracking (`previousSelectedDate`) to detect active date selection (not initial load)
- Added a `LaunchedEffect` that watches `selectedDate` and filters events for the selected date
- Shows snackbar if the filtered events list is empty
- Auto-dismisses after 2.5 seconds using `delay` and `dismiss()`
- Added string resource `text_no_events_on_date` for the message
- Added 5 comprehensive tests covering all branches and lines (80%+ coverage)

The snackbar only appears on active date selection, not on initial page load, and handles edge cases like selecting the same date multiple times.

Fixes #344 

Capture d’écran 2025-12-12 à 01 55 32

### CODE REVIEW

File: app/src/main/java/com/github/se/studentconnect/ui/screen/home/HomeScreen.kt [hunk 1]
```diff
@@ -619,6 +630,40 @@ fun HomeScreen(
             onCalendarDismiss()
           }
         }
+
+        // Check for empty events when date is selected
+        LaunchedEffect(uiState.selectedDate) {
```
Comment (HamzaZbr) [inline app/src/main/java/com/github/se/studentconnect/ui/screen/home/HomeScreen.kt]: Use this as solution if you want: LaunchedEffect(uiState.selectedDate, uiState.events)

**Review by HamzaZbr — CHANGES_REQUESTED** [review]
Thank you Rita for this important fix. I request changes for 2 small issues:
- Please first make sure to fix the Sonar issues: You need only to replace the functions from line 650 to 652 with index accessor ([])
- It's a logic issue: The LaunchedEffect depends only on the selectedDate, but the code depends also on the uiState.events. The problem is that if events are still loading, the filter runs thus it shows the snackbar of empty events and after the events arrive. 
(I'll comment the line with the solution you if you want to use it )

Once you fix this 2 thinks, I'll be happy to merge it !
